lib/babel-istanbul-instrumenter.js

```diff
@@ -0,0 +1,256 @@
+/**
```

Comment:
I'd like to land this PR without this file.  It adds quite a bit of extra weight maintenance wise, and it isn't clear to me how much value it adds.

Can you remove and update?  I'm happy to work through the specific pros/cons in a separate PR after this has landed, but I really don't want to block the other refactors while we hash that out...